force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,39 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+    if (Trigger.isBefore){
+        if(Trigger.isUpdate){
+            Set<Id> accountIds = new Set<Id>();
+            for(Opportunity opp : Trigger.new){
+                if(opp.Amount < 5000) {
+                    opp.addError('Opportunity amount must be greater than 5000');
+                }
+                accountIds.add(opp.AccountId);
+            }
+    
+            List<Contact> contList = [SELECT Id, Name, AccountId, Title FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO'];
+            for(Opportunity opp : Trigger.new){
+                for(Contact c : contList){
+                    if(c.AccountId == opp.AccountId){
+                        opp.Primary_Contact__c = c.Id;
+                        break;
+                    }
```

Comment:
Double nested for loop is not best practice. Try using a map to get all the contactIds by Accounts